app/src/main/java/samples/despotoski/nikola/com/bottomsheetsample/BottomSheetActivity.kt

```diff
@@ -0,0 +1,124 @@
+        }
+
+        val recyclerView = binding.layoutBottomSheet1.recyclerview
+        recyclerView.layoutManager = LinearLayoutManager(this).apply {
+            isAutoMeasureEnabled = true
+        }
+        recyclerView.adapter = ApplicationAdapter(this, listApplications(this))
+        val parentThatHasBottomSheetBehavior = recyclerView.parent.parent as FrameLayout
+        mBottomSheetBehavior = BottomSheetBehavior.from(parentThatHasBottomSheetBehavior)
+        mBottomSheetBehavior?.let {
+            binding.stateText.setText(getStateAsString(it.state))
+            it.setBottomSheetCallback(object : BottomSheetCallback() {
+                override fun onStateChanged(bottomSheet: View, newState: Int) {
+                    binding.stateText.setText(getStateAsString(newState))
+                }
+
+                override fun onSlide(bottomSheet: View, slideOffset: Float) {
+                    binding.offsetText.text = getString(R.string.offset, slideOffset)
+                }
+            })
+        }
+    }
+
+    override fun onBackPressed() {
+        mBottomSheetBehavior?.let {
+            if (it.state != BottomSheetBehavior.STATE_HIDDEN) {
+                it.setState(BottomSheetBehavior.STATE_COLLAPSED)
+            } else {
+                super.onBackPressed()
+            }
+        }
+
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        menuInflater.inflate(R.menu.menu_bottom_sheet, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        val id = item.itemId
+        return if (id == R.id.action_settings) {
+            true
+        } else super.onOptionsItemSelected(item)
+    }
+
+    companion object {
+        fun getStateAsString(newState: Int): Int {
+            when (newState) {
+                BottomSheetBehavior.STATE_COLLAPSED -> return R.string.collapsed
+                BottomSheetBehavior.STATE_DRAGGING -> return R.string.dragging
+                BottomSheetBehavior.STATE_EXPANDED -> return R.string.expanded
+                BottomSheetBehavior.STATE_HIDDEN -> return R.string.hidden
+                BottomSheetBehavior.STATE_SETTLING -> return R.string.settling
+            }
+            return R.string.undefined
+        }
+
+        fun listApplications(context: Context): List<ApplicationInfo> {
+            val flags = PackageManager.GET_META_DATA
+            val installedApps: MutableList<ApplicationInfo> = ArrayList()
+            val pm = context.packageManager
+            val applications = pm.getInstalledApplications(flags)
+            for (appInfo in applications) {
```

Comment:
Maybe simple `.filter{ }` will remove most of the code. 
It will also remove declaring of the `installedApps` by assignment as :

`val installedApps = applicaitons.filter {  thePredicate }`

let me know what do you think.